blob/fileblob/fileblob_test.go

```diff
@@ -440,3 +457,38 @@ func TestListAtRoot(t *testing.T) {
 		t.Fatalf("Expecting an EOF on next item in list, got: %#v", err)
 	}
 }
+
+func TestSkipMetadata(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	dir, err := ioutil.TempDir("", "fileblob*")
+	if err != nil {
+		t.Fatalf("Got error creating temp dir: %#v", err)
+	}
+	defer os.RemoveAll(dir)
+	b := &bucket{dir: dir, opts: &Options{}}
+	bucket := blob.NewBucket(driver.Bucket(b))
+
+	for _, skipMetadata := range [...]bool{false, true} {
+		b.opts.SkipMetadata = skipMetadata
```

Comment:
Modifying the options using the private field is a bit weird (options are supposed to be immutable once you've created the bucket).

How about moving the loop so it's around the whole thing, so you create a new `bucket` for each of the 2 cases, using  the public `fileblob.OpenBucket` (with the `SkipMetadata` option set appropriately).